src/main/java/sirius/biz/storage/layer2/BasicBlobStorageSpace.java

```diff
@@ -975,7 +984,7 @@ public void rename(B blob, String newName) {
         updateBlobName(blob, newName);
 
         blobKeyToFilenameCache.remove(blob.getBlobKey());
-        blobByPathCache.clear();
+        blobByPathCache.removeAll(REMOVE_BY_FILENAME, blob.getFilename());
```

Comment:
**Question:** Shouldn't this be done before AND after the name changes so also the old entry is removed?

Reply:
Of course that is required. Thanks for noticing that!